### ISSUE TITLE
DisallowModule rule misses `Pexp_ident` — pipe calls with no extra arguments bypass lint

### ISSUE THREAD

**Issue by LTibbetts** [plow-technologies/rescript-linter#15]
### Summary

The `DisallowModule` rule fails to detect disallowed module usage when a module function is called via pipe (`->`) with no additional arguments beyond the pipe target. For example, `x->Belt.List.toArray` is not caught, while `x->Belt.List.map(f)` and `Belt.List.toArray(x)` are.

### Reproduction

Given a config with:
```json
{
  "rule": "DisallowModule",
  "options": {
    "disallowed_module": "Belt.List",
    "suggested_module": "List"
  }
},
{
  "rule": "DisallowModule",
  "options": {
    "disallowed_module": "Belt.Array",
    "suggested_module": "Array"
  }
}

```

| Code | Detected? |
|---|---|
| `Belt.List.toArray(myList)` | ✅ Yes |
| `myList->Belt.List.map(x => x + 1)` | ✅ Yes |
| `myList->Belt.List.has(1, (a, b) => a == b)` | ✅ Yes |
| `myList->Belt.List.toArray` | ❌ **No** |
| `myList->Belt.List.length` | ❌ **No** |
| `myList->Belt.List.toArray->Belt.Array.map(x => x)` | ❌ `Belt.List.toArray` missed, `Belt.Array.map` caught |

The pattern is: any pipe call where the disallowed function receives **only** the pipe target (no extra arguments) is missed.

### Root Cause

The linter's bundled ReScript parser does **not** desugar `x->f` the same way `bsc` does. While `bsc -dparsetree` shows both `x->f` and `f(x)` as identical `Pexp_apply` nodes, the linter's parser appears to represent `x->f` (no extra args) as a pipe operator application where `f` is a bare `Pexp_ident` argument — not as `Pexp_apply` of `f`.

In [`DisallowModuleRule.ml`](https://github.com/plow-technologies/rescript-linter/blob/master/lib/rules/DisallowModuleRule.ml), the `lintExpression` handler only matches:

```ocaml
(* Only catches Pexp_apply — function being actively called *)
Pexp_apply {funct= {pexp_desc= Pexp_ident {txt= ident}; ...}; args= _}
```

It never inspects bare `Pexp_ident` nodes, so when a disallowed module function appears as a `Pexp_ident` (passed to the pipe operator rather than being the direct call target), it's invisible to the rule.

### Suggested Fix

Add a `Pexp_ident` case to `lintExpression` in `DisallowModuleRule.ml`:

```ocaml
(* match bare M.function reference, e.g. in x->M.function *)
| {Parsetree.pexp_desc= Pexp_ident {txt= ident; _}; Parsetree.pexp_loc= loc}
  when match extract_module_path ident with
       | Some module_path -> matches_disallowed_module module_path
       | None -> false ->
    Rule.LintError (meta.ruleDescription, loc)
```

### Impact

Any disallowed module function used via pipe with no extra arguments bypasses lint entirely. This affects single-argument functions like `toArray`, `length`, `headExn`, `reverse`, `toFloat`, `toString`, etc. across all modules covered by `DisallowModule` rules.
